src/components/TwoOptionsToggle.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState } from 'react';
+import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
+import { COLORS } from '../styles/themes';
+
+/**
+ * @param {{
+ *   options: Map<'first'|'second', { value: string; onTap: Function; }>;
+ *   defaultOption: 'first'|'second';
+ * }}
+ */
+export const TwoOptionsToggle = ({ options, defaultOption }) => {
+  const [currOption, setCurrOption] = useState(defaultOption);
+  const isActive = (option) => (componentOption) => option === componentOption;
+  const onTap = (option) => {
+    setCurrOption(option);
+    // Execute the callback assigned to the option
+    options[option].onTap();
+  };
+  return (
+    <View style={[styles.wrapper]}>
+      <Option option='first' optionValue={options.first.value} isActive={isActive(currOption)} onTap={onTap} />
+      <Option option='second' optionValue={options.second.value} isActive={isActive(currOption)} onTap={onTap} />
+    </View>
+  );
+};
+
+/**
+ * @param {{
+ *   option: 'first'|'second';
+ *   optionValue: string;
+ *   isActive: (componentOption: string) => boolean;
```

Comment:
Why don't you receive here directly a boolean?

Reply:
Oh, this way the component is able to determine by itself, without external control.

Reply:
And what's the benefit of determining by itself? Why is it better to adding this code complexity instead of receiving `true/false`?

Reply:
Composability. It makes Option component usable by other components. But I see your concern, for now it doesn't have other use.

Reply:
I still don't understand how it's more usable by other components. A simple boolean is as usable as this and the code ir cleaner, which is easier to understand and to maintain in the future.

Reply:
Ok. I will try to make it simple.

Reply:
See it now.